ops/_private/harness.py

```diff
@@ -2225,7 +2227,7 @@ def decorator(target_cls: Any):
 class _TestingConfig(Dict[str, Union[str, int, float, bool]]):
     """Represents the Juju Config."""
 
-    _supported_types = {
+    _supported_types: ClassVar[Dict[str, Any]] = {
```

Comment:
I wonder if ClassVar is correct here.
The field is only mutable because it's a dict.
Semantically, it's not meant to be modified.

Reply:
Yes, I think here it only indicates the attribute is a class variable and should not be set on instances of the class, the emphasis isn't on the variable/mutable part. I made this change so that I could remove the RUF012 rule from the ignore list. I will keep the type unchanged, same for other private class variables.